R/docs.R

```diff
@@ -114,37 +114,58 @@ create_doc_template <- function(experiment, save_dir) {
 #'   set of \code{Experiments}.
 #'
 #' @name render_docs
-#' @description Knits an R Markdown file summarizing the results of an
-#'   \code{Experiment} or set of \code{Experiments}. Outputs an R
-#'   Markdown-generated html file. If \code{experiment} is provided, the results
-#'   are saved in the \code{Experiment}'s root results directory (see
-#'   \code{Experiment$get_save_dir()}). Otherwise, the root results directory is
-#'   taken to be that specified by \code{save_dir}. Note that
-#'   \code{render_docs()} will process and include results from all
-#'   \code{Experiments} found *under* the root directory.
+#' @description Knits and/or writes an R Markdown file summarizing the results of
+#'   an \code{Experiment} or set of \code{Experiments}. This document may
+#'   contain (1) the code corresponding to the \code{DGPs}, \code{Methods},
+#'   \code{Evaluators}, \code{Visualizers}, and \code{Vary Across Parameters}
+#'   from the \code{Experiment}, (2) the results of the \code{Evaluators}
+#'   (typically tables), and (3) the results of the \code{Visualizers}
+#'   (typically figures). Note that \code{render_docs()} will process and
+#'   include results from all \code{Experiments} found *under* the root
+#'   directory. This root directory is determined via
+#'   \code{experiment$get_save_dir()} if \code{experiment} is provided and
+#'   \code{save_dir} otherwise.
 #'
 #' @inheritParams init_docs
-#' @param open If \code{TRUE}, open the R Markdown-generated html file in a
-#'   web browser.
+#' @inheritParams rmarkdown::render
+#' @param write_rmd Logical indicating whether or not to write out the raw
+#'   R Markdown file used to generate the results. If \code{TRUE}, both the
+#'   raw R Markdown file and the rendered R Markdown output are saved to disk.
+#'   If \code{FALSE}, only the rendered R Markdown output is saved. Default is
+#'   \code{FALSE}.
+#' @param output_file The name of the output file. If using \code{NULL}, then
+#'   the output filename will be based on the \code{experiment}'s root results
+#'   directory and the name of the \code{experiment}. Note that the filename
+#'   may include the path to the output file.
+#' @param output_format The R Markdown output format to convert to. Must be an
+#'   object of class `rmarkdown_output_format` (e.g.,
+#'   [rmarkdown::html_document()], [rmarkdown::pdf_document()],
+#'   [rmarkdown::md_document]).
+#' @param output_ext File extension for the rendered R Markdown output document.
```

Comment:
Ah, right I missed that `vthemes::vmodern()` was the default here. Hmm, I'm unsure what this would take, but it might eventually be nice to have `vmodern` as a `theme` yaml arg, like:
```yaml
# rmarkdown
output:
  html_document:
    theme: vmodern
```
or
```yaml
# quarto
format:
  html:
    theme: united
```
And possibly `vmodern` could also be a valid theme for other output formats, like `pdf`. Then maybe we could add an argument `theme` with default `"vmodern"` and have `output_format = "html"` as the default. This would probably be best to consider if/when migrating to `quarto`.

Reply:
Yeah I agree this would be super nice. We should definitely migrate to `quarto` eventually and work on this then.